data_structures/test_stack_legacy.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
```

Comment:
We should probably specify this is using raw pointers and not smart pointers probably?

Reply:
This file is the file with the _old tests_. It uses the same `Stack` as the other tests (i.e. currently with `std::shared_ptr`). In my opinion it could be removed - do you agree on that?

Reply:
Unless it is for some sort of educational purpose, it can be removed. 🙂

Reply:
For educational reasons, I think it is better to remove it. Thanks to that, I can change make `top` and `pop` throw an [`std::invalid_argument`](https://en.cppreference.com/w/cpp/error/invalid_argument) when the stack is empty.